src/app/components/event-card/event-card.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+import { ActivatedRoute } from '@angular/router';
+import { IEvent } from 'src/app/models/event';
+import { CategoryService } from 'src/app/services/category.service';
+import { EventService } from 'src/app/services/event.service';
+
+
+@Component({
+  selector: 'app-event-card',
+  templateUrl: './event-card.component.html',
+  styleUrls: ['./event-card.component.scss']
+})
+export class EventCardComponent implements OnInit {
+  @Input() event: IEvent;
+  categoryIcon: SafeResourceUrl;
+  countdownDays: number;
+  
+  constructor(private categoryService: CategoryService, private eventService: EventService, private sanitizer: DomSanitizer) {}
+
+  ngOnInit(): void {
+    this.initalizeCategoryIcon();
+    this.calculateRemainingDays();
+  }
+
+  initalizeCategoryIcon(): void {
+    this.categoryService.getCategory(this.event.categoryId).subscribe((result) => {
+      this.categoryIcon = this.sanitizer.bypassSecurityTrustResourceUrl(result.result.icon);
```

Comment:
I suggest modifying the service to return only the result object from the API response, this way in the component you can access result.icon